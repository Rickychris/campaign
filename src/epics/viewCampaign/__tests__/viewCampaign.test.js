import React from "react";
import ReactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import ViewCampaign from "../viewCampaign";

describe("viewCampaign  component tests", () => {
  beforeEach(cleanup);
  const viewCampaignTest = (
    <BrowserRouter>
      <ViewCampaign />
    </BrowserRouter>
  );
  it("renders the ViewCampaign component", () => {
    const div = document.createElement("div");
    ReactDom.render(viewCampaignTest, div);
  });

  it("renders the viewCampaign with text correctly", () => {
    const { getByTestId } = render(viewCampaignTest);
    expect(getByTestId("view-campaign")).toHaveTextContent("Campaign List");
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(viewCampaignTest);
    expect(getByTestId("view-campaign")).toMatchSnapshot();
  });
});
