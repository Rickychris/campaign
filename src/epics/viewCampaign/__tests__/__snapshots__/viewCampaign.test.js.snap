// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewCampaign  component tests matches snapshot 1`] = `
<div
  class="root"
  data-testid="view-campaign"
>
  <h1
    class="h1"
  >
    Campaign List
  </h1>
  <h4
    class="h4"
  >
    No Data Found!
  </h4>
  <div
    class="root"
    data-testid="spinner"
  >
    <div
      class="spinner"
    />
  </div>
</div>
`;
